class-two-factor-core.php

```diff
@@ -179,17 +179,25 @@ public static function get_providers() {
 		/**
 		 * For each filtered provider,
 		 */
-		foreach ( $providers as $class => $path ) {
+		foreach ( $providers as $provider_key => $path ) {
 			include_once $path;
 
+			/**
+			 * Filters the classname for a provider.
+			 *
+			 * @param string $provider_key The provider name, which is the default for the Classname.
+			 * @param string $path         The provided provider path to be included.
+			 */
+			$class = apply_filters( 'two_factor_provider_classname', $provider_key, $path );
```

Comment:
```suggestion
			$class = apply_filters( "two_factor_provider_classname_{$provider}", $provider_key, $path );
```

Perhaps it would be worthwhile not having a generic filter, but one that's specific to the provider only.

Reply:
seems neater and less error-prone to have specific filters for each provider.

Reply:
Yeah, that's my thought.

If someone _needs to_ filter all of them, they can fetch the list of keys from the `two_factor_providers` filter and go from there.

Reply:
Added in 51cfc3a7cb5a1f2a2d5892a11be157e9ab95f98a